src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
This implementation should go into `viewer-impl.js`, there's already `messagingOrigin_` there that can be used. Also, `getReferrerUrl` returns a promise, so this implementation is somewhat incorrect. Also, `getReferrerUrl` might be a good idea here, but could you explain why?